post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java

```diff
@@ -131,25 +132,42 @@ protected void processCachedIds() {
                         processTopic(keyTopic, entity, ids, patientRepository::executeStoredProcForPatientIds);
                         break;
                     case INVESTIGATION:
-                        List<InvestigationResult> invData = processTopic(keyTopic, entity, ids, investigationRepository::executeStoredProcForPublicHealthCaseIds);
+                        List<InvestigationResult> invData = processTopic(keyTopic, entity, ids,
+                                investigationRepository::executeStoredProcForPublicHealthCaseIds);
                         ids.forEach(id -> {
                             if (idVals.containsKey(id)) {
-                                processId(id, idVals.get(id), investigationRepository::executeStoredProcForPageBuilder, Entity.CASE_ANSWERS);
+                                processId(id, idVals.get(id),
+                                        investigationRepository::executeStoredProcForPageBuilder, Entity.CASE_ANSWERS);
                                 idVals.remove(id);
                             }
                         });
-                        processTopic(keyTopic, Entity.F_PAGE_CASE, ids, investigationRepository::executeStoredProcForFPageCase);
+                        processTopic(keyTopic, Entity.F_PAGE_CASE, ids,
+                                investigationRepository::executeStoredProcForFPageCase);
                         datamartProcessor.process(invData);
+
+                        /* CNDIT-1584: Notifications has a dependency on Investigation and should be processed
+                        together to eliminate timing issues.*/
+                        Optional<Entry<String, List<Long>>> notificationIds = sortedEntries.stream()
```

Comment:
Notification should ideally go right after investigation. Also, is this processing notification_uids instead of passing the phc_uid set?

Reply:
Fixed!